Assets/Scripts/CameraControl.cs

```diff
@@ -18,12 +19,14 @@ private void DeselectAllBoxes()
         {
             box.GetComponent<Outline>().enabled = false;
         }
+        gameManager.GetComponent<PlayerMovement>().selectedBox = null;
     }
     private void SelectBox(GameObject box)
     {
         DeselectAllBoxes();
         Outline outline = box.GetComponent<Outline>();
         outline.enabled = true;
+        gameManager.GetComponent<PlayerMovement>().selectedBox = box;
```

Comment:
Since the cubes are nested in another game object, select the parent instead of the cube:

```suggestion
        gameManager.GetComponent<PlayerMovement>().selectedBox = box.transform.parent.gameObject;
```